### DISCUSSION TITLE
Using powertools outside Lambda runtime env

### DISCUSSION THREAD

**Discussion by andreaspoldi** [aws-powertools/powertools-lambda-python#3271] [General]
Hi,
 sorry in advance for the dumb question; I was wondering if theres any way to use this library outside Lambda environment context.
There are features like Tracing, Logging and most of all (in my case at least) Batch processing, Zip transformation, S3 streaming etc which are simply awesome within this library, and that I'd like to have also in a container environment.

Did anybody tried this already ? 

Thanks

**Comment by andreaspoldi** [comment]
closing in favour of this https://github.com/aws-powertools/powertools-lambda-python/discussions/3018

**Comment by leandrodamascena** [comment]
Hello @andreaspoldi! Sorry for the delay in responding, yesterday was a busy day!

Have you found the guidance you need? Feel free to reopen this discussion if you need additional help!
